### PR TITLE
webhook: prevent users from creating CheCluster resources

### PR DESCRIPTION
**Pull request by xcoulon**
causes troubles in CRW when multiple `CheCluster` resources co-exist on
the cluster

also, configure webhook logger to produce JSON content
also, move `klogWriter` into its own pkg to share between
operator and webhook

E2E tests: https://github.com/codeready-toolchain/toolchain-e2e/pull/597 

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]

**Review by rajivnathan — APPROVED** [review]
Looks good, just a few minor comments

File: pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	}
+	if requestingUser.GetLabels()[toolchainv1alpha1.ProviderLabelKey] == toolchainv1alpha1.ProviderLabelValue {
+		log.Info("sandbox user is trying to create a CheCluster", "AdmissionReview", admReview)
+		return denyAdmissionRequest(admReview, errors.New("this is a Dev Sandbox enforced restriction. you are trying to create a CheCluster resource, which is not allowed"))
+	}
+	// at this point, it is clear the user isn't a sandbox user, allow request
+	return allowAdmissionRequest(admReview)
+}
+
+func denyAdmissionRequest(admReview v1.AdmissionReview, err error) []byte {
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go]: done in 99c0f1c with a bit of refactoring (moving back both admission webhooks in the same pkg)

File: pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+
+	userv1 "github.com/openshift/api/user/v1"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/admission/v1"
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go]: yes! done in 61b1ece

File: pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajivnathan) [reply pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go]: Nice! 👍

[the remark this answers is not in the document]
Comment (xcoulon) [reply pkg/webhook/validatingwebhook/checluster/validate_checluster_request.go]: thanks! 😃

**Pull request merged**